pallets/subtensor/tests/math.rs

```diff
@@ -0,0 +1,2344 @@
+    let x: Vec<I32F32> = vec![
+        I32F32::from_num(-1.0),
+        I32F32::from_num(10.0),
+        I32F32::from_num(30.0),
+    ];
+    let y: Vec<I32F32> = normalize(&x);
+    assert_vec_compare(
+        &y,
+        &[
+            I32F32::from_num(-0.0256410255),
+            I32F32::from_num(0.2564102563),
+            I32F32::from_num(0.769230769),
+        ],
+        epsilon,
+    );
+    assert_float_compare(sum(&y), I32F32::from_num(1.0), epsilon);
+}
+
+#[test]
+fn test_math_inplace_normalize() {
+    let epsilon: I32F32 = I32F32::from_num(0.0001);
+    let mut x1: Vec<I32F32> = vec![
+        I32F32::from_num(1.0),
+        I32F32::from_num(10.0),
+        I32F32::from_num(30.0),
+    ];
+    inplace_normalize(&mut x1);
+    assert_vec_compare(
+        &x1,
+        &[
+            I32F32::from_num(0.0243902437),
+            I32F32::from_num(0.243902439),
+            I32F32::from_num(0.7317073171),
+        ],
+        epsilon,
+    );
+    let mut x2: Vec<I32F32> = vec![
+        I32F32::from_num(-1.0),
+        I32F32::from_num(10.0),
+        I32F32::from_num(30.0),
+    ];
+    inplace_normalize(&mut x2);
+    assert_vec_compare(
+        &x2,
+        &[
+            I32F32::from_num(-0.0256410255),
+            I32F32::from_num(0.2564102563),
+            I32F32::from_num(0.769230769),
+        ],
+        epsilon,
+    );
+}
+
+#[test]
+fn test_math_inplace_normalize_64() {
+    let epsilon: I64F64 = I64F64::from_num(0.0001);
+    let mut x1: Vec<I64F64> = vec![
+        I64F64::from_num(1.0),
+        I64F64::from_num(10.0),
+        I64F64::from_num(30.0),
+    ];
+    inplace_normalize_64(&mut x1);
+    assert_vec_compare_64(
+        &x1,
+        &[
+            I64F64::from_num(0.0243902437),
+            I64F64::from_num(0.243902439),
+            I64F64::from_num(0.7317073171),
+        ],
+        epsilon,
+    );
+    let mut x2: Vec<I64F64> = vec![
+        I64F64::from_num(-1.0),
+        I64F64::from_num(10.0),
+        I64F64::from_num(30.0),
+    ];
+    inplace_normalize_64(&mut x2);
+    assert_vec_compare_64(
+        &x2,
+        &[
+            I64F64::from_num(-0.0256410255),
+            I64F64::from_num(0.2564102563),
+            I64F64::from_num(0.769230769),
+        ],
+        epsilon,
+    );
+}
+
+#[test]
+fn test_math_vecdiv() {
+    let x: Vec<I32F32> = vec_to_fixed(&[]);
+    let y: Vec<I32F32> = vec_to_fixed(&[]);
+    let result: Vec<I32F32> = vec_to_fixed(&[]);
+    assert_eq!(result, vecdiv(&x, &y));
+
+    let x: Vec<I32F32> = vec_to_fixed(&[0., 1., 0., 1.]);
+    let y: Vec<I32F32> = vec_to_fixed(&[0., 1., 1., 0.]);
+    let result: Vec<I32F32> = vec_to_fixed(&[0., 1., 0., 0.]);
+    assert_eq!(result, vecdiv(&x, &y));
+
+    let x: Vec<I32F32> = vec_to_fixed(&[1., 1., 10.]);
+    let y: Vec<I32F32> = vec_to_fixed(&[2., 3., 2.]);
+    let result: Vec<I32F32> = vec![fixed(1.) / fixed(2.), fixed(1.) / fixed(3.), fixed(5.)];
+    assert_eq!(result, vecdiv(&x, &y));
+}
+
```

Comment:
maybe also check that `vecdiv`'s behavior with division by zero is correct, like:

```suggestion

#[test]
fn test_math_vecdiv_zero_division() {
    let x: Vec<I32F32> = vec_to_fixed(&[1., 2., 3.]);
    let y: Vec<I32F32> = vec_to_fixed(&[0., 0., 0.]);
    let result = vecdiv(&x, &y);
    assert!(result.iter().all(|&val| val.is_nan())); // Expect NaN or error
}

```

Reply:
> maybe also check that `vecdiv`'s behavior with division by zero is correct, like:

This isnt code i added, so I wouldnt want to modify this test